tests/test_models/test_nn/test_mlp.py

```diff
@@ -67,42 +96,49 @@ def test_mlp_make_samples(df_name, request):
             "decoder_real": df[["regressor_float", "regressor_int"]]
             .iloc[encoder_length + decoder_length * i : encoder_length + decoder_length * (i + 1)]
             .values,
+            "decoder_categorical": {
+                column: df[[column]]
+                .iloc[encoder_length + decoder_length * i : encoder_length + decoder_length * (i + 1)]
+                .values
+                for column in cat_columns
+            },
             "decoder_target": df[["target"]]
             .iloc[encoder_length + decoder_length * i : encoder_length + decoder_length * (i + 1)]
             .values,
         }
 
-        assert ts_samples[i].keys() == {"decoder_real", "decoder_target", "segment"}
+        assert ts_samples[i].keys() == {"decoder_real", "decoder_categorical", "decoder_target", "segment"}
         assert ts_samples[i]["segment"] == "segment_1"
         for key in expected_sample:
             np.testing.assert_equal(ts_samples[i][key], expected_sample[key])
 
 
 def test_mlp_forward_fail_nans():
     batch = {
-        "decoder_real": torch.Tensor([[torch.nan, 2, 3], [1, 2, 3], [1, 2, 3]]),
-        "decoder_target": torch.Tensor([[1], [2], [3]]),
+        "decoder_real": torch.Tensor([[[torch.nan, 2, 3], [1, 2, 3], [1, 2, 3]]]),
```

Comment:
Why do we need this change?

Reply:
In the majority of models (may be except nbeast) it is assumed (hardcoded by layers initialisation, for example `batch_first=True` parameter in `nn.RNN`) that we have three dimensions and the first is batch_size.
We implemented embedding features in assumption that we have three dimensions.

Reply:
Do I understand correctly, that there was a mistake in a previous version of this test?

Reply:
Our batch doesn't look like input in previous version of test but it doesn't raise a error because linear layer can get input with and without batch_size dimension
So I can say "yes"